doc/how-to-analyze-haskell-code-performance.rst

```diff
@@ -0,0 +1,124 @@
+How to analyze Haskell performance
+==================================
+
+When a Haskell application is slow or uses too much memory,
+Cabal and the
+`Haskell GHC compiler <https://downloads.haskell.org/ghc/latest/docs/users_guide/>`__
+can help you understand why.
+
+The main steps are to
+
+1. let GHC insert performance measuring code into your application,
+2. run the application to collect a performance report and
+3. visualize that report.
+
+The process of inserting performance measuring code and collecting performance information
+is called "profiling". Cabal makes this easy to do.
+
+Profiling CPU performance
+-------------------------
+
+First, build your application, e.g. ``my-app``, with profiling enabled:
+
+.. code-block:: console
+
+    $ cabal build --enable-profiling --profiling-detail=late my-app
+
+Under the hood, this will pass the corresponding
+`profiling compiler options <https://downloads.haskell.org/ghc/latest/docs/users_guide/profiling.html#compiler-options-for-profiling>`__
+to GHC. The additional Cabal flag ``--profiling-detail=late`` instructs GHC to use
+so-called
+`late-cost-center profiling <https://downloads.haskell.org/ghc/latest/docs/users_guide/profiling.html#ghc-flag--fprof-late>`__
+and insert performance measuring code only after important optimisations
+have been applied to your application code.
+This reduces the performance slow-down of profiling itself and gives you more realistic measurements.
+See the Cabal section on :ref:`profiling options <profiling-options>` for more details.
+
+.. note::
+    You can add further three recommended profiling options to your ``my-app.cabal`` file
+    to control where measuring code is inserted into your application as follows:
+
+    ::
+
+        executable my-app
+          ...
+          ghc-options:
```

Comment:
This is another place where `ghc-prof-options` might make more sense.

Reply:
I just renamed the option to `ghc-prof-options` (i had overlooked the precise spelling). What other place do you mean? I couldn't find any.

Reply:
@Kleidukos mentioned it [here](https://github.com/haskell/cabal/pull/10539#issuecomment-2465305967) although in fact I now think he was referring to this same place, not a different one.